x-pack/metricbeat/module/statsd/server/data_test.go

```diff
@@ -204,9 +204,11 @@ func process(t testing.TB, packets []string, ms *MetricSet) {
 				"client_ip": "127.0.0.1",
 			},
 		}
-		err := ms.processor.Process(udpEvent)
-		require.NoError(t, err)
+		if err := ms.processor.Process(udpEvent); err != nil {
+			return err
+		}
```

Comment:
I see you have moved the error check out of the function, if it was because the line information on errors was pointing here, you could instead use [`t.Helper()`](https://golang.org/pkg/testing/#T.Helper) to mark this function as a test helper so the error is reported from the caller.

Reply:
that's neat! I moved it because i needed it to fail for the SampleRate=0.0 example, so i moved it out and i now check the return value instead